toolkit/tools/internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,56 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/osinfo"
+	"github.com/microsoft/azurelinux/toolkit/tools/pkg/imagecustomizerlib"
+	autoexport "go.opentelemetry.io/contrib/exporters/autoexport"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace/noop"
+)
+
+var shutdownFn func(ctx context.Context) error
+
+func InitTelemetry(disableTelemetry bool) error {
+	if disableTelemetry || os.Getenv("OTEL_EXPORTER_OTLP_ENDPOINT") == "" {
+		logger.Log.Trace("Disabled telemetry collection")
+		otel.SetTracerProvider(noop.NewTracerProvider())
```

Comment:
`GetTracerProvider()` initializes to the no-op provider. So, you don't need to do anything here.